src/components/PopupWithForm.js

```diff
@@ -15,16 +19,21 @@ export default class PopupWithForm extends Popup {
         });
         return this._formValues;
     }
-
+    setLoadingState(isLoading) {
+        if (this._submitButton) {
+          this._submitButton.textContent = isLoading
+          ? "Saving..."
+          : this._defaultButtonText;
+        }
+      }
 
     setEventListener() {
         super.setEventListeners();
         //submit event 
         this._popupForm.addEventListener("submit", (event) => {
             event.preventDefault();
             this._handleFormSubmit(this._getInputValues());
-            // this._popupForm.reset();
-
+            this._popupForm.reset();
         });
     }

```

Comment:
**COULD BE IMPROVED**

You can make a special method for getting a form element from `PopupWithForm` not to **search** it in `index.js`



```js

  getForm() {
    return this._popupForm
  }


```